Runtime/Triangulator.cs

```diff
@@ -173,187 +234,217 @@ public class InputData
-                _ => throw new NotImplementedException()
-            };
-
-            return dependencies;
-        }
-
-        private JobHandle SchedulePCATransformation(JobHandle dependencies)
-        {
-            tmpInputPositions = Input.Positions;
-            Input.Positions = localPositions.AsDeferredJobArray();
-            if (Input.HoleSeeds.IsCreated)
-            {
-                tmpInputHoleSeeds = Input.HoleSeeds;
-                Input.HoleSeeds = localHoleSeeds.AsDeferredJobArray();
-            }
-
-            dependencies = new PCATransformationJob(this).Schedule(dependencies);
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                dependencies = new PCATransformationHolesJob(this).Schedule(dependencies);
-            }
-            return dependencies;
-        }
-
-        private JobHandle SchedulePCAInverseTransformation(JobHandle dependencies)
-        {
-            dependencies = new PCAInverseTransformationJob(this).Schedule(this, dependencies);
-
-            Input.Positions = tmpInputPositions;
-            tmpInputPositions = default;
-
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                Input.HoleSeeds = tmpInputHoleSeeds;
-                tmpInputHoleSeeds = default;
-            }
-
-            return dependencies;
-        }
-
-        private JobHandle ScheduleWorldToLocalTransformation(JobHandle dependencies)
-        {
-            tmpInputPositions = Input.Positions;
-            Input.Positions = localPositions.AsDeferredJobArray();
-            if (Input.HoleSeeds.IsCreated)
-            {
-                tmpInputHoleSeeds = Input.HoleSeeds;
-                Input.HoleSeeds = localHoleSeeds.AsDeferredJobArray();
-            }
-
-            dependencies = new InitialLocalTransformationJob(this).Schedule(dependencies);
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                dependencies = new CalculateLocalHoleSeedsJob(this).Schedule(dependencies);
+        [BurstCompile]
+        struct TriangulationJob: IJob {
+            public Preprocessor preprocessor;
+            public bool validateInput;
+            public bool restoreBoundary;
+            public bool refineMesh;
+            public int sloanMaxIters;
+            public float concentricShellsParameter;
+            public float refinementThresholdArea;
+            public float refinementThresholdAngle;
+            public InputData input;
+            public OutputData output;
+
+            public struct InputData {
+                public NativeArray<float2> Positions;
+
+                [NativeDisableContainerSafetyRestriction]
+                public NativeArray<int> ConstraintEdges;
+
+                [NativeDisableContainerSafetyRestriction]
+                public NativeArray<float2> HoleSeeds;
+            }
+
+            // Markers
+            static readonly ProfilerMarker MarkerPreProcess = new ProfilerMarker("PreProcess");
```

Comment:
I think this will not work on older versions of Unity that don't support this C# syntax. I'm not quite sure what the cutoff is, though. For my use cases, I'll want to be compatible with Unity 2021.3 at least.

Reply:
You are right, however, in `package.json` there is unity 2022.2 defined. Also, support for older unity version is already broken (see e.g. `Edge struct`). This suggestion is not important at the moment. This can be fixed later with or without old version support :)